backend/src/gamestate/game.rs

```diff
@@ -320,28 +292,112 @@ impl Game {
         self.can_player_play(player_name.clone(), &card)?;
 
         let possible_position = self.get_finished_players().len();
-        let player = self
-            .players
-            .iter_mut()
-            .find(|player| player.name() == player_name)
-            .unwrap();
 
-        let mut played_card = player.play_card_by_eq(card)?;
-        if played_card.should_be_black() {
-            if let Some(color) = maybe_new_color {
-                played_card = played_card.morph_black_card(color).unwrap();
+        // scope where player needs to be mutable
+        let played_card = {
+            let player = self
+                .players
+                .iter_mut()
+                .find(|player| player.name() == player_name)
+                .unwrap();
+
+            let mut played_card = player.play_card_by_eq(card)?;
+            if played_card.should_be_black() {
+                if let Some(color) = maybe_new_color {
+                    played_card = played_card.morph_black_card(color).unwrap();
+                }
+            }
+
+            if player.cards().is_empty() {
+                player.set_position(possible_position);
+                // todo!("Send FinishWSMessage");
+            }
+            played_card
+        };
+
+        match played_card.symbol {
+            CardSymbol::Value(_) | CardSymbol::Wild => self.active_cards.clear(),
+            CardSymbol::Reverse => {
+                self.reverse();
+                self.active_cards.clear();
+            }
+            CardSymbol::Draw2 | CardSymbol::Draw4 | CardSymbol::Skip => {
+                self.active_cards.push(played_card.clone()).unwrap();
             }
         }
 
         self.deck.play(played_card);
         // todo!("Send PlayCardWSMessage");
 
-        if player.cards().is_empty() {
-            player.set_position(possible_position);
-            // todo!("Send FinishWSMessage");
+        Ok(())
+    }
+}
+
+pub(super) mod active_cards {
```

Comment:
Můžeme to vytáhnout do vlastního souboru? Případně do vlastního modulu?

Reply:
Však... vlastní modul to je :D

Reply:
Ale jinak asi ano, můžu.

Reply:
Pardon, namísto modulu jsem měl na mysli složku :D

Reply:
Fixed in 494c777b2569bf97ca88907792265f5f088eeb38.